src/sentry/integrations/pagerduty/actions/form.py

```diff
@@ -43,25 +43,23 @@ def __init__(self, *args, **kwargs):
         self.fields["service"].choices = services
         self.fields["service"].widget.choices = self.fields["service"].choices
 
-    def _validate_service(self, service_id: int, integration_id: int | None) -> None:
+    def _validate_service(self, service_id: int, integration_id: int) -> None:
         params = {
             "account": dict(self.fields["account"].choices).get(integration_id),
             "service": dict(self.fields["service"].choices).get(service_id),
         }
 
-        try:
-            service = PagerDutyService.objects.get(id=service_id)
-        except PagerDutyService.DoesNotExist:
-            raise forms.ValidationError(
-                _('The service "%(service)s" does not exist in the %(account)s Pagerduty account.'),
-                code="invalid",
-                params=params,
-            )
-
-        integration = integration_service.get_integration(
-            organization_integration_id=service.organization_integration_id
+        ois = integration_service.get_organization_integrations(
+            integration_id=integration_id,
+            providers=[ExternalProviders.PAGERDUTY.name],
         )
-        if not integration or integration.id != integration_id:
+
+        if not any(
+            pds
+            for oi in ois
+            for pds in oi.config.get("pagerduty_services", [])
+            if pds["id"] == service_id
```

Comment:
is it possible to have more than one match per service_id? If so is this expected/valid behavior?

Reply:
Nope, there should only be one -- the service_id is a global unique identifier from the PagerDutyService object.  In the future, we'll likely generate these by other means when we drop the model all together.